java/vector/src/test/java/org/apache/arrow/vector/ipc/TestArrowFile.java

```diff
@@ -865,4 +875,59 @@ private void write(FieldVector parent, File file, OutputStream outStream) throws
       }
     }
   }
+
+  @Test
+  public void testFileStreamHasEos() throws IOException {
+
+    try (VarCharVector vector1 = newVarCharVector("varchar1", allocator)) {
+      vector1.allocateNewSafe();
+      vector1.set(0, "foo".getBytes(StandardCharsets.UTF_8));
+      vector1.set(1, "bar".getBytes(StandardCharsets.UTF_8));
+      vector1.set(3, "baz".getBytes(StandardCharsets.UTF_8));
+      vector1.set(4, "bar".getBytes(StandardCharsets.UTF_8));
+      vector1.set(5, "baz".getBytes(StandardCharsets.UTF_8));
+      vector1.setValueCount(6);
+
+      List<Field> fields = Arrays.asList(vector1.getField());
+      List<FieldVector> vectors = Collections2.asImmutableList(vector1);
+      VectorSchemaRoot root =  new VectorSchemaRoot(fields, vectors, vector1.getValueCount());
+
+      // write data
+      ByteArrayOutputStream out = new ByteArrayOutputStream();
+      ArrowFileWriter writer = new ArrowFileWriter(root, null, newChannel(out));
+      writer.start();
+      writer.writeBatch();
+      writer.end();
+
+      try (SeekableReadChannel channel = new SeekableReadChannel(
+          new ByteArrayReadableSeekableByteChannel(out.toByteArray()));
+          ArrowFileReader reader = new ArrowFileReader(channel, allocator)) {
+
+        Schema readSchema = reader.getVectorSchemaRoot().getSchema();
+        assertEquals(root.getSchema(), readSchema);
+        assertEquals(1, reader.getRecordBlocks().size());
+
+        ArrowBlock lastBlock = reader.getRecordBlocks().get(0);
```

Comment:
I believe the intent of the change is to ensure files are streamable.  At better test might be to construct an IPC reader that skips over the "magic" file header and padding and verify it reads the batches back successfull.

Reply:
Thanks for your clarify, PR updated.